include/sound/sof/dai.h

```diff
@@ -53,11 +53,13 @@
 #define SOF_DAI_FMT_CLOCK_PROVIDER_MASK	0xf000
 
 /* DAI_CONFIG flags */
-#define SOF_DAI_CONFIG_FLAGS_MASK	0x3
+#define SOF_DAI_CONFIG_FLAGS_MASK	0xFF
 #define SOF_DAI_CONFIG_FLAGS_NONE	(0 << 0) /**< DAI_CONFIG sent without stage information */
 #define SOF_DAI_CONFIG_FLAGS_HW_PARAMS	(1 << 0) /**< DAI_CONFIG sent during hw_params stage */
 #define SOF_DAI_CONFIG_FLAGS_HW_FREE	(2 << 0) /**< DAI_CONFIG sent during hw_free stage */
-#define SOF_DAI_CONFIG_FLAGS_RFU	(3 << 0) /**< not used, reserved for future use */
+#define SOF_DAI_CONFIG_FLAGS_PAUSE     (3 << 0) /**< DAI_CONFIG sent during pause */
```

Comment:
I see, in the firmware the `flags` are not treated as flags:
```c
switch (config->flags & SOF_DAI_CONFIG_FLAGS_MASK) {
case SOF_DAI_CONFIG_FLAGS_HW_PARAMS:
...
case SOF_DAI_CONFIG_FLAGS_HW_FREE:
...
default:
...
}
```

Reply:
the initial code had separate flags but later on we moved to values within a mask, we should have renamed all this...

And now this FLAGS_MASK makes no sense since the member is a u8...

Reply:
PAUSE_PUSH? let's make it explicit and symmetrical with PAUSE_RELEASE.

Reply:
yes, removing the flag would make it easier to use the entire range of 255 values for more flags if needed in the future. I'llchange it.

Reply:
this is no longer relevant with my new changes.